### PR TITLE
Bump Groovy from 2.4.7 to 2.4.21

### PR DESCRIPTION
**Pull request by basil**
Not for review. Filing for testing purposes. Timestamped snapshots built locally (with passing tests!) available at:

https://repo.jenkins-ci.org/artifactory/snapshots/com/cloudbees/groovy-cps/1.33-SNAPSHOT/

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -194,4 +194,10 @@
         </profile>
     </profiles>
 
+  <repositories>
+    <repository>
+      <id>repo.jenkins-ci.org</id>
+      <url>https://repo.jenkins-ci.org/public/</url>
```
Comment (jglick) [inline pom.xml:200]: Out of curiosity, why?

Comment (basil) [reply pom.xml:200]: Try building without this hunk and you will see.

**Comment by jglick** [comment]
Note that due to the DGM processing tool in this repo, this change would have an actual runtime effect, though very minor: https://github.com/apache/groovy/commit/f2f1921d7518312f0e531fde5ce3b6a869597b01

**Comment by jglick** [comment]
Eligible for merge now?

**Comment by basil** [comment]
> Eligible for merge now?

Sure. I pushed a commit undoing the build changes, since those were just a temporary workaround in order for me to do my local testing. What's left is the actual Groovy version bump change. Note that with the build change reverted, the main branch doesn't compile for me, but that's also the case before this PR as well.

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
also #112

**Comment by jglick** [comment]
> with the build change reverted, the main branch doesn't compile for me

Could you file separately please?

We also hope to move this repo to @jenkinsci at which point it should get a `Jenkinsfile` that should catch that sort of thing.

**Comment by basil** [comment]
Sure, though that would require me to actually think about the correct build fix rather than blindly editing POM files like a monkey in front of a typewriter until the thing compiles. I'll put it on the TODO list.

**Comment by jglick** [comment]
I am not sure a monkey placed in front of a typewriter would ever come up with a valid POM.

**Review by car-roll — APPROVED** [review]

**Comment by dwnusbaum** [comment]
The update ended up being merged as part of https://github.com/cloudbees/groovy-cps/pull/118.

**Pull request closed without merging**